python/hidet/graph/ops/definitions/reduce/reduce.py

```diff
@@ -70,23 +68,158 @@ def reduce_fcompute(*reduce_indices):
             },
         )
 
-    def implement_cuda(self, working_dir: str) -> IRModule:
-        # pylint: disable=import-outside-toplevel
-        from ...schedules import cuda_schedule_reduce_by_default, cuda_schedule_reduce_by_warp_reduce
-
-        if self.inputs[0].type.dtype.name == 'float64':
-            return NotImplemented  # use auto-scheduler
-
-        if not builtins.all(is_constant(dim) for dim in self.inputs[0].shape):
-            return NotImplemented
+    def allow_epilogue(self) -> bool:
+        # return False
+        rank = len(self.inputs[0].shape)
+        if rank - 1 in self.dims:  # pylint: disable=simplifiable-if-statement
+            # use self.cuda_schedule_reduce_by_warp
+            return True
+        else:
+            # use self.cuda_schedule_reduce_by_default
+            return False
```

Comment:
Just looked at the `self.cuda_schedule_reduce_by_default(...)` implementation, seems it is okay to fuse prologue.